src/core/load-fragment.c

```diff
@@ -532,156 +532,149 @@ int config_parse_exec(
 
         e += ltype;
 
+        /* FIXME: ExecStart=<empty> clears the list, but ExecStart=<whitespace>
+         * doesn't, they should behave the same.  */
         if (isempty(rvalue)) {
                 /* An empty assignment resets the list */
                 *e = exec_command_free_list(*e);
                 return 0;
         }
 
-        /* We accept an absolute path as first argument, or
-         * alternatively an absolute prefixed with @ to allow
-         * overriding of argv[0]. */
-        for (;;) {
+        rvalue += strspn(rvalue, WHITESPACE);
+        p = rvalue;
+
+        do {
                 int i;
-                const char *word, *state, *reason;
-                size_t l;
+                _cleanup_strv_free_ char **n = NULL;
+                _cleanup_free_ ExecCommand *nce = NULL;
+                _cleanup_free_ char *path = NULL;
+                _cleanup_free_ char *firstword = NULL;
+                char *f;
                 bool separate_argv0 = false, ignore = false;
 
-                path = NULL;
-                nce = NULL;
-                n = NULL;
+                semicolon = false;
 
-                rvalue += strspn(rvalue, WHITESPACE);
+                r = unquote_first_word(&p, &firstword, UNQUOTE_CUNESCAPE|UNQUOTE_UNESCAPE_RELAX);
+                if (r < 0)
```

Comment:
Yes, I think that would help users to spot problems in their unit files. Just use log_syntax(), as that will already include the current line and file that is parsed, and augment that with whatever caused the error (in this case, 'p').